src/opengl/platform/rendercontext_ogl_win.cpp

```diff
@@ -0,0 +1,113 @@
+
+    // Initialize extensions
+    // Get the device context
+    m_deviceContext = GetDC(hwndFake);
+    // Set a temporary pixel format
+    errorCode = SetPixelFormat(m_deviceContext, 1, &m_pixelFormat);
+    // Create temporary render context
+    m_renderContext = wglCreateContext(m_deviceContext);
+    // Set the temporary rendering context as the current rnedering context for this
+    // window
+    errorCode = wglMakeCurrent(m_deviceContext, m_renderContext);
+    // Initialize the OpenGL extensions, we need a temp context to do so
+    GLenum err = glewInit();
+    if (err != GLEW_OK) {
+        std::cerr << "Direwolf: Failed to initialize extensions!";
+    }
+
+    // Release the temporary render context now that the extensions have loaded
+    wglMakeCurrent(nullptr, nullptr);
+    wglDeleteContext(m_renderContext);
+
+    // Release device context
+    ReleaseDC(hwndFake, m_deviceContext);
+
+    // Get the device context again
+    m_hwnd = static_cast<HWND>(platformData.windowHandle);
+    m_deviceContext = GetDC(m_hwnd);
+
+    int32_t attrs[] = {WGL_SAMPLE_BUFFERS_ARB,
+                       0,
+                       WGL_SAMPLES_ARB,
+                       0,
+                       WGL_SUPPORT_OPENGL_ARB,
+                       true,
+                       WGL_PIXEL_TYPE_ARB,
+                       WGL_TYPE_RGBA_ARB,
+                       WGL_DRAW_TO_WINDOW_ARB,
+                       true,
+                       WGL_DOUBLE_BUFFER_ARB,
+                       true,
+                       WGL_COLOR_BITS_ARB,
+                       32,
+                       WGL_DEPTH_BITS_ARB,
+                       24,
+                       WGL_STENCIL_BITS_ARB,
+                       8,
+                       0};
+
+    int pixelFormat1[1];
+    unsigned int formatCount;
+    // Query for a pixel format that fits the attributes we want
+    errorCode = wglChoosePixelFormatARB(m_deviceContext, attrs, NULL, 1, pixelFormat1,
+                                        &formatCount);
+    // If the gpu/display can handle the desired pixel format we set it as the current one
+    errorCode = SetPixelFormat(m_deviceContext, pixelFormat1[0], &m_pixelFormat);
+    int attributeList[5];
+    // Set the 4.0 version of OpenGL in the attribute list.
+    attributeList[0] = WGL_CONTEXT_MAJOR_VERSION_ARB;
+    attributeList[1] = 4;
+    attributeList[2] = WGL_CONTEXT_MINOR_VERSION_ARB;
+    attributeList[3] = 0;
+    // Null terminate the attribute list.
+    attributeList[4] = 0;
+    m_renderContext = wglCreateContextAttribsARB(m_deviceContext, 0, attributeList);
+    if (!m_renderContext) {
+        std::cerr << "DireWolf: Failed to initialize OpenGL Context\n";
+    }
+    errorCode = wglMakeCurrent(m_deviceContext, m_renderContext);
+    // Get the name of the video card.
+	auto vendorString = static_cast<const unsigned char*>(glGetString(GL_VENDOR));
+    auto versionString = static_cast<const unsigned char*>(glGetString(GL_VERSION));
```

Comment:
tabs vs spaces and `auto`.. but I guess we said `auto` is fine when type is specified on the rhs.
However, seems like this string is only used once. Do we need the variable? Can it be called from within the `cout` without the cast?

Reply:
can try without, the whole file needs refactor though. future task/issue